theano/tensor/opt.py

```diff
@@ -1460,7 +1460,7 @@ def apply(self, fgraph):
 # As local_shape_to_shape_i is not in fast_compile,
 # dont register this opt in fast_compile.
 theano.compile.mode.optdb.register('ShapeOpt', ShapeOptimizer(),
-                                   0.1, 'fast_run')
+                                   0.1, 'fast_run', 'fast_compile')
```

Comment:
Yeah, i've discussed this with @nouiz . He asked me to keep it here like this and do that work in another PR